manifests/base/templates/services.yaml

```diff
@@ -1,3 +1,34 @@
+{{- if .Values.global.remotePilotAddress }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-pilot
+  namespace: {{ .Values.global.istioNamespace }}
+spec:
+  ports:
+  - port: 15010
+    name: grpc-xds # direct
+  - port: 15011
+    name: https-xds # mTLS or non-mTLS depending on auth setting
+  - port: 8080
+    name: http-legacy-discovery # direct
+  - port: 15012
+    name: http-istiod
+  - port: 15014
+    name: http-monitoring
+  clusterIP: None
```

Comment:
Q - do we purposefully not want selector for this service?

Reply:
Yes, its a headless / selector-less service (see https://kubernetes.io/docs/concepts/services-networking/service/#without-selectors). It isn't backed by a pod in this cluster.

Reply:
A bit confused why this is headless but istiod-remote is not? In the description you said istio-remote IS headless

Reply:
istiod remote is also headless because it doesn't have any selector.

Reply:
Its missing `clusterIP: None`. I'll add it.

Reply:
@ayj istio-remote in 1.4 should have had `clusterIP: None` already - as the implementation was based upon a headless service. Do you mean that you missed that in this PR, or instead, that 1.4 has an issue on this point?

Reply:
Refresh. The istiod-remote service has `ClusterIP: None` now. I forgot to add it in the initial commit.